File: testdata/fixtures/bench/python/vulnerable/a02_hardcoded_secrets.py
# Source: CWE-798 - Hardcoded credentials in application code
# Expected: BATOU-SEC-001 (Hardcoded Password), BATOU-SEC-002 (API Key), BATOU-SEC-004 (Connection String)
# OWASP: A02:2021 - Cryptographic Failures (Hardcoded Secrets)

import boto3
import pymysql

DATABASE_PASSWORD = "p@ssw0rd123!"
AWS_ACCESS_KEY = "AKIAIOSFODNN7EXAMPLE"
AWS_SECRET_KEY = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
API_KEY = "sk-proj-abc123def456ghi789"

def get_db_connection():
    conn_string = "mysql://admin:p@ssw0rd123!@db.internal.example.com:3306/production"
    return pymysql.connect(
        host='db.internal.example.com',
        user='admin',
        password=DATABASE_PASSWORD,
        database='production'
    )

def get_s3_client():
    return boto3.client(
        's3',
        aws_access_key_id=AWS_ACCESS_KEY,
        aws_secret_access_key=AWS_SECRET_KEY,
    )

def call_external_api(endpoint: str):
    import requests
    headers = {'Authorization': f'Bearer {API_KEY}'}
    return requests.get(endpoint, headers=headers)

File: batou-rules/testdata/fixtures/bench/swift/vulnerable/a02_hardcoded_secrets.swift
// Source: CWE-798 - Hardcoded API keys and credentials in Swift
// Expected: BATOU-SWIFT
// OWASP: A02:2021 - Cryptographic Failures (Hardcoded Secrets)

import Foundation

let API_KEY = "sk_live_4eC39HqLyjWDarjtT1zdp7dc"
let DATABASE_PASSWORD = "SuperSecretPass123!"
let AWS_SECRET_ACCESS_KEY = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"

func connectToAPI() {
    let url = URL(string: "https://api.example.com/data")!
    var request = URLRequest(url: url)
    request.setValue("Bearer \(API_KEY)", forHTTPHeaderField: "Authorization")
}

func connectToDB() {
    let connStr = "postgresql://admin:\(DATABASE_PASSWORD)@db.example.com/prod"
    print(connStr)
}
